src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -22,24 +34,67 @@ public class FindCommandParser implements Parser<FindCommand> {
+        if (argMultimap.getValue(PREFIX_NAME).isEmpty() && argMultimap.getValue(PREFIX_PHONE).isEmpty()
+                && argMultimap.getValue(PREFIX_EMAIL).isEmpty() && argMultimap.getValue(PREFIX_DEPARTMENT).isEmpty()
+                && argMultimap.getValue(PREFIX_ROLE).isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
-        } else if (typeOfPerson.equals(FindEmployeeCommand.ARGUMENT_WORD)) {
-            return new FindEmployeeCommand(new NameContainsKeywordsPredicate(keywords));
+        }
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            List<String> nameKeywords = Arrays.stream(
+                    argMultimap.getAllValues(PREFIX_NAME).get(0).split("\\s+")).toList();
+            NameContainsKeywordsPredicate nameContainsKeywordsPredicate =
+                    new NameContainsKeywordsPredicate(nameKeywords);
+            predicateContainer.addNameContainsKeywordsPredicate(nameContainsKeywordsPredicate);
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            List<String> phoneKeywords = Arrays.stream(
+                    argMultimap.getAllValues(PREFIX_PHONE).get(0).split("\\s+")).toList();
+            PhoneContainsKeywordsPredicate phoneContainsKeywordsPredicate =
+                    new PhoneContainsKeywordsPredicate(phoneKeywords);
+            predicateContainer.addPhoneContainsKeywordsPredicate(phoneContainsKeywordsPredicate);
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            List<String> emailKeywords = Arrays.stream(
+                    argMultimap.getAllValues(PREFIX_EMAIL).get(0).split("\\s+")).toList();
+            EmailContainsKeywordsPredicate emailContainsKeywordsPredicate =
+                    new EmailContainsKeywordsPredicate(emailKeywords);
+            predicateContainer.addEmailContainsKeywordsPredicate(emailContainsKeywordsPredicate);
+        }
+        if (argMultimap.getValue(PREFIX_DEPARTMENT).isPresent()) {
+            List<String> departmentKeywords = Arrays.stream(
+                    argMultimap.getAllValues(PREFIX_DEPARTMENT).get(0).split("\\s+")).toList();
+            DepartmentContainsKeywordsPredicate departmentContainsKeywordsPredicate =
+                    new DepartmentContainsKeywordsPredicate(departmentKeywords);
+            predicateContainer.addDepartmentContainsKeywordsPredicate(departmentContainsKeywordsPredicate);
+        }
+        if (argMultimap.getValue(PREFIX_ROLE).isPresent()) {
+            List<String> roleKeywords = Arrays.stream(
+                    argMultimap.getAllValues(PREFIX_ROLE).get(0).split("\\s+")).toList();
+            RoleContainsKeywordsPredicate roleContainsKeywordsPredicate =
+                    new RoleContainsKeywordsPredicate(roleKeywords);
+            predicateContainer.addRoleContainsKeywordsPredicate(roleContainsKeywordsPredicate);
+        }
```

Comment:
Perhaps lines 56-90 could be abstracted into a function that dynamically handles the differences? The statements inside every if statement look the same and perhaps the methods for predicateContainer could be abstracted into one function inside the class to handle the input types?

Reply:
Updated. FindCommandParser will use a PredicateContainer static method to extract the values out from the argumentMultimap. Add more methods in PredicateContainer to set the Predicate fields using the argumentMultimap.